kratos/utilities/variable_utils.h

```diff
@@ -1531,6 +1534,120 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
         const Vector& rPositions
         );
 
+    /// @brief Check whether a @ref Node, @ref Element, @ref Condition, @ref ProcessInfo, or @ref ModelPart stores a value for the provided @ref Variable.
+    /// @param rEntity @ref Node, @ref Element, @ref Condition, @ref ProcessInfo, or @ref ModelPart to check.
+    /// @param rVariable to check.
+    template <Globals::DataLocation TLocation, class TEntity, class TValue>
+    static bool HasValue(const TEntity& rEntity, const Variable<TValue>& rVariable)
+    {
+        if constexpr (TLocation == Globals::DataLocation::NodeHistorical) {
+            static_assert(std::is_same_v<TEntity,Node>);
+            return rEntity.SolutionStepsDataHas(rVariable);
+        } else {
+            static_assert(std::is_same_v<TEntity,Node>
+                          || std::is_same_v<TEntity,Element>
+                          || std::is_same_v<TEntity,Condition>
+                          || std::is_same_v<TEntity,ProcessInfo>
+                          || std::is_same_v<TEntity,ModelPart>);
+            return rEntity.Has(rVariable);
+        }
+    }
+
+    /// @brief Fetch the value of a variable stored in an entity.
+    /// @param rEntity @ref Node, @ref Element, @ref Condition, @ref ProcessInfo, or @ref ModelPart to fetch data from.
+    /// @param rVariable @ref Variable to fetch the value of.
+    template <Globals::DataLocation TLocation, class TEntity, class TValue>
+    static std::conditional_t<std::is_integral_v<TValue> || std::is_floating_point_v<TValue>,
```

Comment:
```suggestion
    static std::conditional_t<std::is_arithmetic_v<TValue>,
```
https://en.cppreference.com/w/cpp/types/is_arithmetic